main.go

```diff
@@ -97,23 +86,32 @@ func main() {
 	}
 
 	pullRequestHandler := &PRCreateHandler{
-		ClientCreator:           cc,
-		releaseManagerAuthToken: *releaseManagerAuthToken,
-		releaseManagerURL:       *releaseManagerURL,
-		messageTemplate:         *messageTemplate,
-		repoFilters:             *repoFilter,
+		ClientCreator:                   cc,
+		releaseManagerMetricsMiddleware: clientMetricsMiddleware(prometheusRegistry, "release-manager")(http.DefaultTransport),
+		releaseManagerAuthToken:         *releaseManagerAuthToken,
+		releaseManagerURL:               *releaseManagerURL,
+		messageTemplate:                 *messageTemplate,
+		repoFilters:                     *repoFilter,
 	}
 
 	webhookHandler := githubapp.NewDefaultEventDispatcher(githubappConfig, pullRequestHandler)
 
-	server.Mux().Handle(pat.Post("/webhook/github/bot"), webhookHandler)
+	// Create http server
+	mux := http.NewServeMux()
+	mux.Handle(*githubWebhookRoute, inboundMetricsMiddleware(prometheusRegistry, webhookHandler))
+	mux.Handle(*metricsRoute, promhttp.Handler())
 
-	// Start is blocking
-	err = server.Start()
-	if err != nil {
-		logger.Error().Msgf("Failed to serve: %v", err)
-		os.Exit(1)
-		return
+	// Middleware
+	httpHandler := loggerMiddleware(func(msg string, m map[string]interface{}) {
+		httpLogger.Info().Fields(m).Msg(msg)
+	}, mux)
+	httpHandler = addContextMiddleware(logger, httpHandler)
+
+	s := http.Server{
+		Addr:    fmt.Sprintf(":%d", httpServerConfig.Port),
```

Comment:
This could make use of [`net.JoinHostPort`](https://godoc.org/net#JoinHostPort) instead. That will make it IP v6 compatible.

```suggestion
		Addr:    net.JoinHostPort("", strconv.Itoa(httpServerConfig.Port)),
```

Reply:
Awesome idea